components/script/dom/urlsearchparams.rs

```diff
@@ -135,9 +141,15 @@ impl URLSearchParamsMethods for URLSearchParams {
     }
 
     // https://url.spec.whatwg.org/#dom-urlsearchparams-has
-    fn Has(&self, name: USVString) -> bool {
+    fn Has(&self, name: USVString, value: Option<USVString>) -> bool {
         let list = self.list.borrow();
-        list.iter().any(|&(ref k, _)| k == &name.0)
+        list.iter().any(|&(ref k, ref v)| {
+            if let Some(value) = &value {
+                k == &name.0 && v == &value.0
+            } else {
+                k == &name.0
+            }
+        })
```

Comment:
I think this could be rewritten as a match statement:

```rust
           match &value {
               Some(value)  => k == &name.0 && v == &value.0
               None =>  k == &name.0
            }
```